manticore/utils/rlp.py

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+
+import collections.abc
+
+def int_to_bytes(n):
+    return n.to_bytes((n.bit_length() + 7) // 8, 'big')
+
+def rlp_encode(item):
+    if item is None or item == 0:
+        return b'\x80'
+    elif isinstance(item, str):
+        return rlp_encode(item.encode('utf8'))
+    elif isinstance(item, bytearray) or isinstance(item, bytes):
+        if len(item) == 1 and item[0] < 0x80:
+            return item
+        else:
+            return encode_length(len(item), 0x80) + item
+    elif isinstance(item, collections.abc.Sequence):
+        output = b''.join(map(rlp_encode, item))
+        return encode_length(len(output), 0xC0) + output
+    elif isinstance(item, int) or isinstance(item, long):
+        return rlp_encode(int_to_bytes(item))
+    else:
+        raise Exception("Cannot encode object of type %s" % type(item).__name__)
+
+def encode_length(length, offset):
+    if length < 56:
```

Comment:
Some docstring e.g. why the magic number 56?

Reply:
It's all from the Ethereum RLP specification. I'll add a link to it.